### PR TITLE
WIP feat: connect agent to the API using gRPC

### PR DESCRIPTION
**Pull request by moul**

### CODE REVIEW

**Comment by Z-a-r-a-k-i** [comment]
For local testing, you need to start a mariadb pathwar docker instance (go into deployments/platform-dev then docker-compose up db)
Then start pathwar api server
Now the grpc calls should be done properly and you are ready to debug them

File: deployments/platform-dev/docker-compose.yml [hunk 1]
```diff
@@ -78,10 +78,12 @@ services:
     volumes:
       - db_data:/var/lib/mysql
     environment:
-      - MYSQL_ROOT_PASSWORD
+      - MYSQL_ROOT_PASSWORD=uns3cur3
```
Comment (moul) [inline deployments/platform-dev/docker-compose.yml]: this file is not meant to be used by developers but for deployments

on deployment servers, we always use .env files to configure unique env vars

please revert back this file, but feel free to update/create docker-compose files outside of deployments/ directory to make the maintainers' life easier

Comment (Z-a-r-a-k-i) [reply deployments/platform-dev/docker-compose.yml]: okay sure

**Comment by moul** [comment]
closing in favor of #409

**Pull request closed without merging**